torchgeo/datasets/ssl4eo.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Self-Supervised Learning for Earth Observation."""
+
+import os
+from typing import Callable, Dict, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import check_integrity, extract_archive
+
+
+class SSL4EOS12(NonGeoDataset):
+    """SSL4EO-S12 dataset.
+
+    The `Self-Supervised Learning for Earth Observation - Sentinel-1/2
+    <https://github.com/zhu-xlab/SSL4EO-S12>`_ dataset is a large-scale mutilmodal
+    multitemporal dataset for unsupervised/self-supervised pre-training in Earth
+    observation.
+
+    The dataset consists of unlabeled patch triplets (Sentinel-1 dual-pol SAR,
+    Sentinel-2 top-of-atmosphere multispectral, Sentinel-2 surface reflectance
+    multispectral) from 251079 locations across the globe, each patch covering
+    2640mx2640m and including four seasonal time stamps.
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/2211.07044
+
+    .. note::
+
+       This dataset can be downloaded using:
+
+       .. code-block:: console
+
+          $ RSYNC_PASSWORD=m1660427.001 rsync -av rsync://m1660427.001@dataserv.ub.tum.de/m1660427.001/ .
+
+       The dataset is about 1.5 TB when compressed and 3.7 TB when uncompressed, and
+       takes roughly 36 hrs to download, 1 hr to checksum, and XX hrs to extract.
```

Comment:
Popcorn not included. BYOP.

Still trying to extract the full dataset, will update XX once I know the full time it takes.